### PR TITLE
[ACR] Add commands for Conditional Access policy (2022-02-01-preview API)

### PR DESCRIPTION
**Pull request by cegraybl**
**Related command**
new commands
`az acr config authentication-as-arm show -r <registry>`
`az acr config authentication-as-arm update -r <registry> --status [enabled/disabled]`

**Description**
Add new commands `az acr config authentication-as-arm show` `az acr config authentication-as-arm update`
Upgrade SDK version to 10.0.0
Update all live tests
Lock network-rule command group to API version 2021-08-01-preview

**Testing Guide**
`az acr config authentication-as-arm show -r <registry>`
Shows the current configuration of the registry's policy for authentication using ARM tokens with the registry. `enabled` means both ACR's and ARM's tokens can be used for authentication, `disabled` means only ACR's tokens can be used for authentication

`az acr config authentication-as-arm update -r <registry> --status [enabled/disabled]`
Updates the value of the registry's policy, the allowed values are `enabled` and `disabled`

**History Notes**
[ACR] `az acr config authentication-as-arm show`: Add new command to support showing the configured 'Azure AD authenticate as ARM' policy
[ACR] `az acr config authentication-as-arm update`: Add new command to support updating 'Azure AD authenticate as ARM' policy

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [X] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [X] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [X] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

File: src/azure-cli-core/azure/cli/core/profiles/_shared.py [hunk 1]
```diff
@@ -292,7 +293,7 @@ def default_api_version(self):
         ResourceType.MGMT_SERVICEBUS: '2021-06-01-preview',
         ResourceType.MGMT_IOTHUB: '2019-07-01-preview',
         ResourceType.MGMT_DATABOXEDGE: '2019-08-01',
-        ResourceType.MGMT_CONTAINERREGISTRY: '2019-05-01',
```
Comment (mabenedi) [inline src/azure-cli-core/azure/cli/core/profiles/_shared.py:295]: are we sure we wanted to bump this one?

Comment (cegraybl) [reply src/azure-cli-core/azure/cli/core/profiles/_shared.py]: spoke offline, it is better if we keep this azurestack hybrid one the current GA version

**Comment by yonzhan** [comment]
ACR

**Comment by cegraybl** [comment]
@zhoxing-ms the tests are passing now, please help me with a review.
Thanks in advance.

File: src/azure-cli/azure/cli/command_modules/acr/_params.py [hunk 2]
```diff
@@ -126,6 +126,10 @@ def load_arguments(self, _):  # pylint: disable=too-many-statements
         c.argument('force', help='Overwrite the existing tag of the image to be imported.', action='store_true')
         c.argument('no_wait', help="Do not wait for the import to complete and return immediately after queuing the import.", action='store_true')
 
+    with self.argument_context('acr config authentication-as-arm') as c:
+        c.argument('registry_name', options_list=['--registry', '-r', c.deprecate(target='-n', redirect='-r', hide=True), c.deprecate(target='--name', redirect='--registry', hide=True)])
+        c.argument('status', help="Indicates whether authentication-as-arm is enabled.", arg_type=get_enum_type(PolicyStatus))
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/acr/_params.py:131]: Please use the first-person voice in the help message
```suggestion
        c.argument('status', help="Indicate whether authentication-as-arm is enabled.", arg_type=get_enum_type(PolicyStatus))
```

Comment (cegraybl) [reply src/azure-cli/azure/cli/command_modules/acr/_params.py]: done

File: src/azure-cli/azure/cli/command_modules/acr/_help.py [hunk 3]
```diff
@@ -70,6 +70,32 @@
...
+helps['acr config authentication-as-arm update'] = """
+type: command
+short-summary: Update 'Azure AD authenticate as ARM' policy for an Azure Container Registry.
+examples:
+  - name: Disable 'Azure AD authenticate as ARM' policy for an Azure Container Registry, so only ACR audienced tokens can be used for authentication
+    text: >
+        az acr config authentication-as-arm update -r MyRegistry --status Disabled
+  - name: Enable 'Azure AD authenticate as ARM' policy for an Azure Container Registry, it will allow both ACR and ARM audienced tokens to be used for authentication
+    text: >
+        az acr config authentication-as-arm update -r MyRegistry --status Enable
```
[the remark this answers is not in the document]
Comment (cegraybl) [reply src/azure-cli/azure/cli/command_modules/acr/_help.py]: done

**Review by mabenedi — APPROVED** [review]
LGTM

**Review by wangzelin007 — COMMENTED** [review]
LGTM

**Review by wangzelin007 — APPROVED** [review]

**Comment by wangzelin007** [comment]
Hi @cegraybl,
I can't find any code about `Deprecate params subnet and vnet-name on network-rule command group`, can you explain?

**Comment by cegraybl** [comment]
@wangzelin007 @zhoxing-ms  the deprecation notice is a mistake from my part, this PR only locks the commands to the old preview API and introduces a new API version.
The actual deprecation will be handled by @mabenedi in [PR 22959](https://github.com/Azure/azure-cli/pull/22959)

**Comment by wangzelin007** [comment]
Hi @cegraybl,
I removed this line based on your feedback: `[ACR] Deprecate params subnet and vnet-name on network-rule command group`
Hi @zhoxing-ms,
Please help review this PR~

File: src/azure-cli/azure/cli/command_modules/acr/_docker_utils.py [hunk 4]
```diff
@@ -123,16 +125,23 @@ def _get_aad_token_after_challenge(cli_ctx,
...
 
+    scope = None
+    if use_acr_audience:
+        logger.debug("Using ACR audience token for authentication")
+        scope = "https://{}.azure.net".format(ACR_AUDIENCE_RESOURCE_NAME)
+
     # this might be a cross tenant scenario, so pass subscription to get_raw_token
     subscription = get_subscription_id(cli_ctx)
-    creds, _, tenant = profile.get_raw_token(subscription=subscription)
+    creds, _, tenant = profile.get_raw_token(subscription=subscription,
```
Comment (susanshi) [inline src/azure-cli/azure/cli/command_modules/acr/_docker_utils.py:143]: are we only getting the ACR token if arm token auth is disabled?

**Comment by zhoxing-ms** [comment]
> [ACR] Add new commands az acr config authentication-as-arm show az acr config authentication-as-arm update, used for conditional access
  [ACR] Bump azure-mgmt-containerregistry version to 10.0.0
  [ACR] Lock network-rule command group to API version 2021-08-01-preview

@cegraybl Since our history notes only write descriptions that users can perceive, so I have removed the above two descriptions that are not related to UX. 
In addition, I also refine the description of those new commands. Please see if the new description meets your expectations.

**Review by zhoxing-ms — APPROVED** [review]

**Comment by cegraybl** [comment]
@zhoxing-ms the history notes look good, thanks for the help.

**Pull request merged**